### ISSUE TITLE
iotDataManager connection attempt never executes callback

### ISSUE THREAD

**Issue by matthew-ely** [aws-amplify/aws-sdk-ios#4950]
**Describe the bug**
Attempting to follow AWS SDK standalone sample code on the Amplify docs. The IoTDataManager never executes the callback nor gives any log indicating it is trying to connect.

**To Reproduce**
1. Follow Amplify docs sample code: https://docs.amplify.aws/sdk/pubsub/getting-started/q/platform/ios/#aws-iot 
2. Attempt to endpoint

**Observed Behavior**
Absolutely no logs.

**Expected Behavior**
I expect the IoTDataManager to at least ATTEMPT to connect to the endpoint.

**Stack Trace**
N/A see observed behavior

**Code Snippet**
// Initialize the AWSIoTDataManager with the configuration
let iotEndPoint = AWSEndpoint(
    urlString: "wss://MY_ENDPOINT-ats.iot.us-gov-west-1.amazonaws.com/mqtt")
let iotDataConfiguration = AWSServiceConfiguration(
    region: AWSRegionType.USGovWest1,
    endpoint: iotEndPoint,
    credentialsProvider: AWSMobileClient.default()
)

AWSIoTDataManager.register(with: iotDataConfiguration!, forKey: "IoTManager")
let iotDataManager = AWSIoTDataManager(forKey: "IoTManager")

**Unique Configuration**
I am attempting to make a connection with an iPhone as an IoT device authenticating with preexisting certificate and private key.

**Areas of the SDK you are using (AWSMobileClient, Cognito, Pinpoint, IoT, etc)?**
AWSMobileClient, IoT

**Environment(please complete the following information):**
 - SDK Version: 2.33.4
 - Dependency Manager: CocoaPods
 - Swift Version : 5
 - Xcode Version: 14.3.1

**Device Information (please complete the following information):**
 - Device: iPhone 14 Pro Simulator, iPhone X physical
 - iOS Version: 16.4

**Comment by ruisebas** [comment]
Hi @matthew-ely, thanks for opening this issue.

Just to clarify, did you also follow the steps in the [Establish Connection](https://docs.amplify.aws/sdk/pubsub/working-api/q/platform/ios/#establish-connection) guide?

The snippet you provided is only creating the service client for your configuration and registering it under the `"IoTManager"` key.

**Comment by matthew-ely** [comment]
Yes, I accidentally omitted that snippet. I tried using the plain 'connect' function as well as the 'connectUsingALPN' function and there was no feedback or log that it was even attempting to connect at all.

**Comment by ruisebas** [comment]
Thanks for confirming. The `statusCallback` is only invoked when a new `AWSIoTMQTTStatus` is received, but if the connection fails it will not be invoked. This could be the reason you are not getting the callback. 

All `connect...` functions return a `Bool` that indicates if the connection was successful.

However, you should be able to get more details in the logs. Don't forget to add a logger, for example if you just want to log to the Xcode console:
```swift 
AWSDDLog.add(AWSDDTTYLogger.sharedInstance)
```
If you still don't get enough information, try enabling verbose logging:
```swift
AWSDDLog.sharedInstance.logLevel = .verbose
```
